swift-extensions/UISwitchExtensions.swift

```diff
@@ -15,21 +15,39 @@ extension UISwitch {
             viewModel = value
             guard let toggleSwitchViewModel = value else { return }
 
-            self.addTarget(self, action: #selector(UISwitch.onValueChanged), for: .valueChanged)
-            observe(toggleSwitchViewModel.isOn) { [weak self] (value: Bool) in
+            observe(PublisherExtensionsKt.distinctUntilChanged(toggleSwitchViewModel.isOn)) { [weak self] (value: Bool) in
                 self?.isOn = value
             }
 
-            bind(toggleSwitchViewModel.enabled, \UISwitch.isEnabled)
+            bind(toggleSwitchViewModel.isEnabled, \UISwitch.isEnabled)
+            
+            observe(toggleSwitchViewModel.toggleSwitchAction) {[weak self] (value: ViewModelAction) in
+                guard let self = self else { return }
+                self.removePreviousRegisteredTapAction()
+                if value != ViewModelAction.Companion().None {
```

Comment:
Why do you require that you have a `toggleSwitchAction` 
The UI should always update the common with when his state changes.

Reply:
Since the absolute truth resides in the repository of the viewmodel integrator we opted for a switch that doesn't necessairly change the UI, instead an action that goes through business logic will determine if the toggle should be switched or not based on a repository value update and therefore a new value being published and observed by the viewmodel.